lib/db/set.go

```diff
@@ -285,21 +371,47 @@ func (s *FileSet) ListDevices() []protocol.DeviceID {
 // database.
 func DropFolder(ll *Lowlevel, folder string) {
 	db := newInstance(ll)
-	db.dropFolder([]byte(folder))
-	db.dropMtimes([]byte(folder))
-	db.dropFolderMeta([]byte(folder))
+	if err := db.dropFolder([]byte(folder)); backend.IsClosed(err) {
+		return
+	} else if err != nil {
+		panic(err)
+	}
+	if err := db.dropMtimes([]byte(folder)); backend.IsClosed(err) {
+		return
+	} else if err != nil {
+		panic(err)
+	}
+	if err := db.dropFolderMeta([]byte(folder)); backend.IsClosed(err) {
+		return
+	} else if err != nil {
+		panic(err)
+	}
 
 	// Also clean out the folder ID mapping.
-	db.folderIdx.Delete([]byte(folder))
+	if err := db.folderIdx.Delete([]byte(folder)); backend.IsClosed(err) {
+		return
+	} else if err != nil {
+		panic(err)
```

Comment:
In theory all of this should be in a "transaction"?

Reply:
In principle... In practice that looks very infections once I start pulling the threads, almost the same as removing the top-level write operations completely. Let me see what I can do.